automation/run-tests.sh

```diff
@@ -58,32 +59,49 @@ function dump_network_info {
 
 function install_nmstate {
     docker_exec '
-      cd /workspace/nmstate &&
       rpm -ivh `./packaging/make_rpm.sh|tail -1 || exit 1`
     '
 }
 
 function run_tests {
-    docker_exec "
-      cd /workspace/nmstate &&
-      pytest \
-        --verbose --verbose \
-        --log-level=DEBUG \
-        --log-date-format='%Y-%m-%d %H:%M:%S' \
-        --log-format='%(asctime)s %(filename)s:%(lineno)d %(levelname)s %(message)s' \
-        --durations=5 \
-        --cov=\$($PYTHON_SITE_PATH_CMD)/libnmstate \
-        --cov=\$($PYTHON_SITE_PATH_CMD)/nmstatectl \
-        --cov-report=html:htmlcov-integ \
-        --cov-report=term \
-        tests/integration \
-    ${nmstate_pytest_extra_args}"
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_LINT ];then
+        docker_exec 'tox -e flake8,pylint'
+    fi
+
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_UNIT_PY27 ];then
+        docker_exec 'tox -e check-py27'
+    fi
+
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_UNIT_PY36 ];then
+        docker_exec 'tox -e check-py36'
+    fi
+
+    if [ $TEST_TYPE == $TEST_TYPE_ALL ] || \
+       [ $TEST_TYPE == $TEST_TYPE_INTEG ];then
+        docker_exec "
+          cd /workspace/nmstate &&
+          pytest \
+            --verbose --verbose \
+            --log-level=DEBUG \
+            --log-date-format='%Y-%m-%d %H:%M:%S' \
+            --log-format='%(asctime)s %(filename)s:%(lineno)d %(levelname)s %(message)s' \
+            --durations=5 \
+            --cov=\$($PYTHON_SITE_PATH_CMD)/libnmstate \
```

Comment:
Why do we need to the full path here instead of just `libnmstate`?

Reply:
The pytest is using rpm installed libnmstate instead of folder in source folder. If using the `libnmstate` only, the pytest coverage will report zero coverage.

Reply:
Thank you. I see. If we can assume that nmstate will be installed only for one python version in one running container, it seems that this can be simplified using a shell glob (does not need `PYTHON_SITE_PATH_CMD` anymore):

```
diff --git a/automation/run-tests.sh b/automation/run-tests.sh
index f6a1cf0..8164f12 100755
--- a/automation/run-tests.sh
+++ b/automation/run-tests.sh
@@ -95,8 +95,8 @@ function run_tests {
             --log-date-format='%Y-%m-%d %H:%M:%S' \
             --log-format='%(asctime)s %(filename)s:%(lineno)d %(levelname)s %(message)s' \
             --durations=5 \
-            --cov=\$($PYTHON_SITE_PATH_CMD)/libnmstate \
-            --cov=\$($PYTHON_SITE_PATH_CMD)/nmstatectl \
+            --cov /usr/lib/python*/site-packages/libnmstate \
+            --cov /usr/lib/python*/site-packages/nmstatectl \
             --cov-report=html:htmlcov-integ \
             --cov-report=term \
             tests/integration \
```
And this might also again allow to move this into tox.ini (there are currently only outdated integration test environments that should probably be removed).

Reply:
Good point. Yes, there is only one python version of nmstate been installed in all tests.
I have created issue #285 to trace this in another PR.